rinja_parser/src/expr.rs

```diff
@@ -188,7 +189,28 @@ impl<'a> Expr<'a> {
     expr_prec_layer!(band, shifts, token_bitand);
     expr_prec_layer!(shifts, addsub, alt((tag(">>"), tag("<<"))));
     expr_prec_layer!(addsub, muldivmod, alt((tag("+"), tag("-"))));
-    expr_prec_layer!(muldivmod, filtered, alt((tag("*"), tag("/"), tag("%"))));
+    expr_prec_layer!(muldivmod, conv, alt((tag("*"), tag("/"), tag("%"))));
+
+    fn conv(start: &'a str, level: Level) -> ParseResult<'a, WithSpan<'a, Self>> {
+        let (i, lhs) = Self::filtered(start, level)?;
+        let (i, kw) = ws(opt(keyword("as")))(i)?;
+        if kw.is_none() {
+            return Ok((i, lhs));
+        }
+
+        let (j, target) = opt(identifier)(i)?;
+        let target = target.unwrap_or_default();
+        match target {
+            "bool" | "f32" | "f64" | "i8" | "i16" | "i32" | "i64" | "i128" | "isize" | "u8"
+            | "u16" | "u32" | "u64" | "u128" | "usize" => {
```

Comment:
IMHO any other conversion should be done in rust code. You should never need to write e.g. `{{ x as Box<dyn Error> }}` in a templating language.

Reply:
Yes, let's keep it simple. :+1: